scripts/job.pbs

```diff
@@ -80,7 +71,23 @@ teardown()
 	log_echo "  $CHECKPOINT_FILE"
 	cp -p "$TEMP_DIR/checkpoint.chk" "$CHECKPOINT_FILE"
 
-	log_echo "Teardown complete, job done ;)"
+	log_echo "Teardown complete"
+}
+
+fchk()
+{
+	log_echo "Formatting checkpoint file:"
+	log_echo "  $FORMATTED_FILE"
+	
+	# creates directory for fchk files
+	mkdir -p $PBS_O_WORKDIR/fchkpoints/"
+	formchk "$CHECKPOINT_FILE" "$FORMATTED_FILE"
```

Comment:
Why not place this in the main function? The purpose of setup -> main -> teardown is so that files are only copied back to the user dir _after_ all work is complete. This is more efficient for the computer and has the benefit of not leaving corrupted files in the user dir if the job were to terminate early.